compiler/rustc_mir/src/interpret/memory.rs

```diff
@@ -625,9 +630,6 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
             // Need to make a copy, even if `get_global_alloc` is able
             // to give us a cheap reference.
             let alloc = Self::get_global_alloc(memory_extra, tcx, id, /*is_write*/ true)?;
-            if alloc.mutability == Mutability::Not {
-                throw_ub!(WriteToReadOnly(id))
-            }
```

Comment:
Surprisingly, this turns out not to be redundant. I will need to investigate that.

Reply:
Ah, it *is* redundant, this just changes the error message in a Stacked Borrows test.